synapse/handlers/events.py

```diff
@@ -61,6 +61,11 @@ def get_stream(self, auth_user_id, pagin_config, timeout=0,
         If `only_keys` is not None, events from keys will be sent down.
         """
 
+        if room_id:
+            blocked = yield self.store.is_room_blocked(room_id)
+            if blocked:
+                raise SynapseError(403, "This room has been blocked on this server")
```

Comment:
for future ref: prefer AuthError?

Reply:
Ah yes, I just copied the existing stuff for consistency, but yup.